org.nanuc.Axolotl.yml

```diff
@@ -39,168 +29,74 @@ finish-args:
-        x-checker-data:
-          type: git
-          tag-pattern: "^v([\\d.]+)$"
-
-  - name: astilectron-bundler-astilectron
-    buildsystem: simple
-    build-options:
-      env:
-        BUNDLER_CACHE: /app/lib/astibundler/cache
-    build-commands:
-      - install -Dm 644 astilectron-0.56.0.zip ${BUNDLER_CACHE}/astilectron-0.56.0.zip
-    sources:
-      - type: file
-        url: https://github.com/asticode/astilectron/archive/v0.56.0.zip
-        sha256: 2725492843edaef0530543c041c5c7d9e43d4163f5c192fc8da11ce89fcaa0f5
-        dest-filename: astilectron-0.56.0.zip
+          tag-pattern: ^v([\d.]+)$
+          # BUG https://github.com/flathub/flatpak-external-data-checker/issues/294
+          #version-scheme: semantic
 
-  - name: astilectron-bundler-electron-amd64
+  - name: axolotl
     buildsystem: simple
-    only-arches:
-      - x86_64
     build-options:
+      # Add the node bin directory.
+      append-path: "/usr/lib/sdk/node18/bin:/run/build/axolotl/flatpak-node/vite:/usr/lib/sdk/rust-stable/bin"
       env:
-        BUNDLER_CACHE: /app/lib/astibundler/cache
+        # Set the cache directory, used to find Electron and some other helper
+        # tools.
+        # (The directory format is: /run/build/MODULE_NAME/flatpak-node/cache)
+        XDG_CACHE_HOME: /run/build/axolotl/flatpak-node/cache
+        # Set the npm cache directory, used by npm to find its package metadata.
+        npm_config_cache: /run/build/axolotl/flatpak-node/npm-cache
+        # Sets the directory where Node is located so that npm won't download the headers.
+        npm_config_nodedir: /usr/lib/sdk/node18
+        # Cargo config
+        CARGO_HOME: /run/build/axolotl/cargo
+        # Cargo offline mode
+        CARGO_NET_OFFLINE: "true"
     build-commands:
-      - install -Dm 644 electron-linux-amd64-20.2.0.zip ${BUNDLER_CACHE}/electron-linux-amd64-20.2.0.zip
-    sources:
-      - type: file
-        url: https://github.com/electron/electron/releases/download/v20.2.0/electron-v20.2.0-linux-x64.zip
-        sha256: c0e3522de34819b838f4a35ddf30c6283c61be1bb8dff02089cda3f641938aad
-        dest-filename: electron-linux-amd64-20.2.0.zip
+      # Install the packages from our offline cache.
+      - npm --prefix=axolotl-web --offline install
 
-  - name: astilectron-bundler-electron-arm64
-    buildsystem: simple
-    only-arches:
-      - aarch64
-    build-options:
-      env:
-        BUNDLER_CACHE: /app/lib/astibundler/cache
-    build-commands:
-      - install -Dm 644 electron-linux-arm64-20.2.0.zip ${BUNDLER_CACHE}/electron-linux-arm64-20.2.0.zip
-    sources:
-      - type: file
-        url: https://github.com/electron/electron/releases/download/v20.2.0/electron-v20.2.0-linux-arm64.zip
-        sha256: 1d2dd05f060786eae5ce687794aa071df8a5f5733b7b2efa1481de8db75d7738
-        dest-filename: electron-linux-arm64-20.2.0.zip
+      # Build axolotl-web
+      - npm --prefix=axolotl-web --offline run build
```

Comment:
Thanks for the proposal! Really appreciate that someone picks up the work :)
The idea here was not to repeat the build commands again, but to re-use the Makefile commands from the axolotl repo.

For example here: This should be make build-axolotl-web.

The axolotl build should be make build-axolotl.

Reply:
I think the `--offline` is important here.

As I could reduce the Flatpak manifest to only a few commands, I think this should be fine. Furthermore, I checked in the same manifest in the Axolotl repo which is built in every PR to make sure nothing breaks.

Reply:
We could set it per call, or we could set it as it was done before:

```yaml
  - name: axolotl
    build-options:
      env:
        npm_config_offline: "true"
```

Reply:
as far as I know, that is the same - just rather than copying and pasting, we do it for the whole module

Reply:
See the doc here for example: https://docs.flatpak.org/en/latest/electron.html

Reply:
Thanks for the additional explanation. Done.